packages/sdk/src/TemplateProvider.ts

```diff
@@ -6,51 +6,91 @@ import * as fs from 'fs'
+        const templateDirectory = this.LGTemplateDirectory()
+        if (templateDirectory === null) {
             return null
         }
 
-        let templateString = JSON.stringify(template)
-        let argumentNames = this.GetArgumentNames(templateString)
-
-        // Substitute argument values
-        for (let argumentName of argumentNames) {
-            let renderedActionArgument = templateArguments.find(a => a.parameter == argumentName)
-            if (renderedActionArgument) {
-                templateString = templateString.replace(new RegExp(`{{${argumentName}}}`, 'g'), renderedActionArgument.value ?? '')
-            }
+        const renderingParameters = {}
+        // Copy parameters defined in the action to our rendering parameters.
+        for (const templateArgument of templateArguments) {
+            renderingParameters[templateArgument.parameter] = templateArgument.value
         }
-        templateString = this.RemoveEmptyArguments(templateString)
-        return JSON.parse(templateString)
+        // Copy entities in memory to our rendering parameters.
+        entityDisplayValues.forEach((value: string, key: string) => { renderingParameters[key] = value })
+
+        // Load the template.  Currently, we assume that each lg file only has one template.
+        const lgFilename = path.join(templateDirectory, templateName + ".lg")
+        const engine = new TemplateEngine().addFile(lgFilename)
+        const output = engine.evaluateTemplate(engine.templates[0].name, renderingParameters)
+        return ActivityFactory.createActivity(output)
     }
 
     public static GetTemplates(): Template[] {
-        let templates: Template[] = []
-        let files = this.GetTemplatesNames()
-        for (let file of files) {
-            let fileContent = this.GetTemplate(file)
-
-            // Clear submit check (will the set by extracting template variables)
-            this.hasSubmitError = false
-            let tvs = this.UniqueTemplateVariables(fileContent)
+        const files = this.GetTemplatesNames()
+        if (files.length === 0) {
+            return []
+        }
+        const templateDirectory = this.LGTemplateDirectory()
+        if (templateDirectory === null) {
+            throw new Error("Could not find valid template directory")
+        }
+        const templates: Template[] = []
+        for (const file of files) {
+            const fileName = path.join(templateDirectory, `${file}.lg`)
+            const engine = new TemplateEngine().addFile(fileName)
+            let templateBody = ''
+            if (file.includes('AdaptiveCard')) {
+                templateBody = engine.templates[1].body
+                templateBody = templateBody.replace("- ```\r\n", "").replace("```", "")
+                for (const symbol of engine.templates[1].parameters) {
```

Comment:
The reason is that the format of other types of cards (e.g., hero, video, audio) is totally different with adaptivecard. For example, it looks like:
[HeroCard 
        title=Cheese gromit!
        subtitle=@{type}
        text=This is some text describing the card, it's cool because it's cool
        image=https://memegenerator.net/img/instances/500x/73055378/cheese-gromit.jpg
        buttons=Option 1| Option 2| Option 3
]
This format could not be directly parsed and rendered by AdaptiveCardViewer.